samples/csharp_dotnetcore/14.nlp-with-dispatch/ConfigurationCredentialProvider.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class ConfigurationCredentialProvider : SimpleCredentialProvider
```

Comment:
For "Future Proofing" I would have this DIRECTLY implement ICredentialProvider and Contain / DelegateTo the SimpleCredentialProvider. That way we can - in the future - change the inheritance tree w/o breaking binary comparability of the Class.

Laugh if you will, but I really like this simple class. Makes the DI system and ASP.Net Integration so much cleaner! I'm glad you had a chance to add it in.